libraries/Filter/LowPassFilter2p.cpp

```diff
@@ -17,13 +17,12 @@ DigitalBiquadFilter<T>::DigitalBiquadFilter() {
 
 template <class T>
 T DigitalBiquadFilter<T>::apply(const T &sample, const struct biquad_params &params) {
-    if(is_zero(params.cutoff_freq) || is_zero(params.sample_freq)) {
+    if(!is_positive(params.cutoff_freq) || !is_positive(params.sample_freq)) {
         return sample;
     }
 
     if (!initialised) {
         reset(sample, params);
-        initialised = true;
```

Comment:
Don't need to set `initialised` because the `reset` call does it for us.

https://github.com/ArduPilot/ardupilot/blob/f72bfcc373b0a61f327cccb0dc5812ecee17db8c/libraries/Filter/LowPassFilter2p.cpp#L45-L48